hoot-core/src/main/cpp/hoot/core/conflate/DiffConflator.cpp

```diff
@@ -127,42 +132,96 @@ void DiffConflator::apply(OsmMapPtr& map)
   {
     _matchFactory.createMatches(_pMap, _matches, _bounds);
   }
-  LOG_DEBUG("Match count: " << _matches.size());
-  LOG_TRACE(SystemInfo::getMemoryUsageString());
-
+  LOG_INFO(
+    "\tFound: " << StringUtils::formatLargeNumber(_matches.size()) <<
+    " Differential Conflation matches.");
   double findMatchesTime = timer.getElapsedAndRestart();
   _stats.append(SingleStat("Find Matches Time (sec)", findMatchesTime));
   _stats.append(SingleStat("Number of Matches Found", _matches.size()));
   _stats.append(SingleStat("Number of Matches Found per Second",
     (double)_matches.size() / findMatchesTime));
+  OsmMapWriterFactory::writeDebugMap(_pMap, "after-matching");
 
   // Use matches to calculate and store tag diff. We must do this before we
   // create the map diff, because that operation deletes all of the info needed
   // for calculating the tag diff.
-  MapProjector::projectToWgs84(_pMap);
   _calcAndStoreTagChanges();
 
-  // We have matches. Here's what we are going to do: because our _pMap contains
-  // all of input1 and input2, we are going to delete everthing that belongs to
-  // a match pair. Then we will delete all remaining input1 items... leaving us
-  // with the differential that we want.
-  for (std::vector<const Match*>::iterator mit = _matches.begin(); mit != _matches.end(); ++mit)
-  {
-    std::set< std::pair<ElementId, ElementId> > pairs = (*mit)->getMatchPairs();
+  // We're eventually getting rid of all matches from the output, but in order to make the road
+  // snapping work correctly we'll get rid of secondary elements in matches first.
+  _removeMatches(Status::Unknown2);
 
-    for (std::set< std::pair<ElementId, ElementId> >::iterator pit = pairs.begin();
-         pit != pairs.end(); ++pit)
-    {
-      RecursiveElementRemover(pit->first).apply(_pMap);
-      RecursiveElementRemover(pit->second).apply(_pMap);
-    }
+  if (ConfigOptions().getDifferentialSnapUnconnectedRoads())
```

Comment:
Road snapping implemented as an optional post conflate op in the diff conflator.  It *has* to happen before diff conflate dumps the ref matches for the diff, which is why I'm not adding it to the post conflate ops as would normally be done for this type of change.